projects/dialogue_bias/util.py

```diff
@@ -0,0 +1,216 @@
+        counts_path = os.path.join(baby_name_folder, f'yob{year:d}.txt')
+        with open(counts_path) as f:
+            for line in f:
+                name, gender, count_string = line.split(',')
+                count = int(count_string.rstrip())
+                baby_name_counts_by_gender[name][gender] += count
+
+    return baby_name_counts_by_gender
+
+
+def get_tzioumis_name_lists(
+    tzioumis_data_path: str,
+    race_gender_name_lists: Dict[str, List[str]],
+) -> Dict[str, List[str]]:
+    """
+    Get race/ethnicity name lists from the Tzioumis work.
+    """
+    percent_df = load_tzioumis_data(tzioumis_data_path)
+    name_lists = {}
+    for race_ethnicity in RACES_ETHNICITIES_WITH_NAMES:
+        name_lists[race_ethnicity] = get_race_ethnicity_name_list_given_tzioumis_data(
+            percent_df=percent_df,
+            race_gender_name_lists=race_gender_name_lists,
+            race_ethnicity=race_ethnicity,
+        )
+    return name_lists
+
+
+def load_tzioumis_data(tzioumis_data_path: str) -> pd.DataFrame:
+    """
+    Load Tzioumis data from
+    https://dataverse.harvard.edu/dataset.xhtml?persistentId=doi%3A10.7910%2FDVN%2FTYJKEZ
+    (Paper: https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5839157/)
+    """
+    print(f'Loading Tzioumis data from {tzioumis_data_path}')
+    percent_df = (
+        pd.read_excel(tzioumis_data_path, sheet_name='Data')[
+            lambda df: df['firstname'] != 'ALL OTHER FIRST NAMES'
+        ]
+        .assign(
+            firstname=lambda df: df['firstname'].apply(lambda s: s[0] + s[1:].lower())
+        )
+        .set_index('firstname')
+    )
+    return percent_df
+
+
+def get_race_ethnicity_name_list_given_tzioumis_data(
+    percent_df: pd.DataFrame,
+    race_gender_name_lists: Dict[str, List[str]],
+    race_ethnicity: str,
+) -> List[str]:
+    """
+    Given input Tzioumis data (percent_df) and a set of names split by gender and
+    race/ethnicity from Milkman et al.
+
+    (2012), Caliskan et al. (2017), and Guo and Caliskan (2020)., get a list of names
+    for the given race/ethnicity.
+    """
+
+    # Params
+    tzioumis_to_race_gender_mapping = {'hispanic': 'his', 'white': 'ea', 'black': 'aa'}
+
+    # Determine which names are most commonly of the specified race/ethnicity (i.e.
+    # plurality), and pick the 200 of those that have the most observations for that
+    # ethnicity
+    this_ethnicity_column = f'pct{race_ethnicity}'
+    percent_columns = [f'pct{race_eth}' for race_eth in RACES_ETHNICITIES]
+    max_percent_series = percent_df[percent_columns].max(axis=1)
+    percent_plurality_names_df = (
+        percent_df[lambda df: df[this_ethnicity_column] == max_percent_series]
+        .assign(
+            obs_of_this_ethnicity=lambda df: df['obs'] * df[this_ethnicity_column] / 100
+        )
+        .sort_values('obs_of_this_ethnicity', ascending=False)
+    )
+    tzioumis_plurality_names = percent_plurality_names_df.iloc[
+        :200
```

Comment:
Not sure I follow, why 200? Can you please add a comment to clarify?

Reply:
Semi-arbitrary, to prevent the lists of names from being too big - adding a comment